### DISCUSSION TITLE
Gradle Maven deploy failing with 422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by gohlinka2** [community/community#26328] [Packages]
<p>Hi,<br>
we use Github Packages for deploying our Kotlin Multiplatform library to a private repo. Recently, I’ve been having trouble publishing a new version, the Gradle <code>publish</code> task fails with:</p>
<pre><code class="lang-auto">Execution failed for task ':coreMechanics:publishAndroidDebugPublicationToGithubPackagesRepository'.
&gt; Failed to publish publication 'androidDebug' to repository 'GithubPackages'
   &gt; Could not PUT 'https://maven.pkg.github.com/fireballrpg/fireball-multiplatform/fireball/coreMechanics-android-debug/1.6.4/coreMechanics-android-debug-1.6.4.aar'. Received status code 422 from server: Unprocessable Entity
</code></pre>
<p>I have tried the following, all of these resulted in the same error:</p>
<ul>
<li>Verified that the problem isn’t specific to one package (the task publishes multiple artifacts automatically generated by Gradle)</li>
<li>Tried rolling back changes from the new version and publishing the previous version (1.6.3) as 1.6.4</li>
<li>Tried publishing the new version as the previous one (1.6.3) - which, IIRC, was previously failing with something like “can’t publish a new package with an existing version”</li>
<li>Tried using different versions - 1.4.0-rc0, 1.6.5, 1.7.0</li>
</ul>
<p>This may be related to</p><aside class="quote quote-modified" data-post="1" data-topic="119670">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/maven-deploy-fails-with-http-422-unprocessable-entity/119670">Maven deploy fails with HTTP 422 Unprocessable Entity</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-packages/43"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #005cc5;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Any questions related to GitHub Packages and how to manage your packages; upload, download, and delete. Happy to see questions about our help docs and the core set of clients and services we support but also questions about configuring and using alternate clients are welcome.">GitHub Packages</span></a>
  </div>
  <blockquote>
    Hi, 
Trying to mvn deploy a multi-module maven project to GHPM (snapshot, not release), but I get the following error on the first file upload: 
[DEBUG] Using transporter WagonTransporter with priority -1.0 for https://maven.pkg.github.com/ttomsu/spring-cloud-gcp
[DEBUG] Using connector BasicRepositoryConnector with priority 0.0 for https://maven.pkg.github.com/ttomsu/spring-cloud-gcp with username=ttomsu, password=***
Downloading from ttomsu-spring-gcp-new: https://maven.pkg.github.com/ttomsu/s…
  </blockquote>
</aside>
<p>
although I haven’t managed to make any use of it for my case and also it seems to be resolved.</p>
<p>The last time I verified this was working was September 30.</p>
<p>POM of the failing package:</p>
<pre><code class="lang-auto">&lt;?xml version="1.0" encoding="UTF-8"?&gt;
&lt;project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"&gt;
  &lt;!-- This module was also published with a richer model, Gradle metadata,  --&gt;
  &lt;!-- which should be used instead. Do not delete the following line which  --&gt;
  &lt;!-- is to indicate to Gradle or any Gradle module metadata file consumer  --&gt;
  &lt;!-- that they should prefer consuming it instead. --&gt;
  &lt;!-- do_not_remove: published-with-gradle-metadata --&gt;
  &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
  &lt;groupId&gt;fireball&lt;/groupId&gt;
  &lt;artifactId&gt;coreMechanics-android-debug&lt;/artifactId&gt;
  &lt;version&gt;1.6.4&lt;/version&gt;
  &lt;packaging&gt;aar&lt;/packaging&gt;
  &lt;dependencies&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;com.squareup.sqldelight&lt;/groupId&gt;
      &lt;artifactId&gt;runtime-jvm&lt;/artifactId&gt;
      &lt;version&gt;1.3.0&lt;/version&gt;
      &lt;scope&gt;compile&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlin&lt;/groupId&gt;
      &lt;artifactId&gt;kotlin-stdlib-common&lt;/artifactId&gt;
      &lt;version&gt;1.3.72&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;com.github.aakira&lt;/groupId&gt;
      &lt;artifactId&gt;napier&lt;/artifactId&gt;
      &lt;version&gt;1.3.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlinx&lt;/groupId&gt;
      &lt;artifactId&gt;kotlinx-serialization-runtime-common&lt;/artifactId&gt;
      &lt;version&gt;0.20.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.kodein.di&lt;/groupId&gt;
      &lt;artifactId&gt;kodein-di-erased-jvm&lt;/artifactId&gt;
      &lt;version&gt;6.5.5&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlinx&lt;/groupId&gt;
      &lt;artifactId&gt;kotlinx-coroutines-core-common&lt;/artifactId&gt;
      &lt;version&gt;1.3.5-native-mt&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;com.squareup.sqldelight&lt;/groupId&gt;
      &lt;artifactId&gt;coroutines-extensions-jvm&lt;/artifactId&gt;
      &lt;version&gt;1.3.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;co.touchlab&lt;/groupId&gt;
      &lt;artifactId&gt;stately-common-jvm&lt;/artifactId&gt;
      &lt;version&gt;1.0.2&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlin&lt;/groupId&gt;
      &lt;artifactId&gt;kotlin-stdlib&lt;/artifactId&gt;
      &lt;version&gt;1.3.72&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;com.github.aakira&lt;/groupId&gt;
      &lt;artifactId&gt;napier-android&lt;/artifactId&gt;
      &lt;version&gt;1.3.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlinx&lt;/groupId&gt;
      &lt;artifactId&gt;kotlinx-serialization-runtime&lt;/artifactId&gt;
      &lt;version&gt;0.20.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;com.squareup.sqldelight&lt;/groupId&gt;
      &lt;artifactId&gt;android-driver&lt;/artifactId&gt;
      &lt;version&gt;1.3.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
      &lt;groupId&gt;org.jetbrains.kotlinx&lt;/groupId&gt;
      &lt;artifactId&gt;kotlinx-coroutines-android&lt;/artifactId&gt;
      &lt;version&gt;1.3.5-native-mt&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
  &lt;/dependencies&gt;
&lt;/project&gt;

</code></pre>
<p>Let me know if I can provide any more info and thanks a lot in advance for any help.</p>

**Comment by fabian** [comment]
<p>Seeing the same issue:</p>
<aside class="quote quote-modified" data-post="4" data-topic="133921">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/adding-mvn-artifacts-with-classifiers/133921/4">Adding MVN Artifacts with classifiers</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-packages/43"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #005cc5;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Any questions related to GitHub Packages and how to manage your packages; upload, download, and delete. Happy to see questions about our help docs and the core set of clients and services we support but also questions about configuring and using alternate clients are welcome.">GitHub Packages</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/jcansdale">@jcansdale</a> Sorry for hijacking this thread but we are also just recently started seeing issues with our MVN packages which worked before. Both SNAPSHOT and sources have been known to cause issues in the past with GitHub Packages so I suspect this is related: 
Error:  Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.7:deploy (default-deploy) on project […]: Failed to deploy artifacts: Could not transfer artifact […]:jar:2.13.0 from/to github (https://maven.pkg.github.com/[…])…
  </blockquote>
</aside>

**Comment by zd-dalibor** [comment, marked as answer]
<aside class="quote no-group" data-username="gohlinka2" data-post="1" data-topic="137299">
<div class="title">
<div class="quote-controls"></div>
 gohlinka2:</div>
<blockquote>
<p><code>&lt;artifactId&gt;coreMechanics-android-debug&lt;/artifactId&gt;</code></p>
</blockquote>
</aside>
<p>artifacId should be lowercase to be able to push to github packages</p>

**Reply by ghost** [reply]
It worked for me, Thanks!

**Reply by okumujustine** [reply]
Thank you, this worked for me

**Comment by gohlinka2** [comment]
<p>Going to try it, but not sure if this is the solution as the previous packages were published with this <code>artifactId</code> just fine. Thanks anyway, going to update once I try it.</p>

**Comment by gohlinka2** [comment]
<p>Okay, so it went through when I changed the <code>artifactId</code>.<br>
Not sure why it was working previously and suddenly stopped, anyway it is a solution so thanks a lot!</p>

**Comment by mikenakis** [comment]
<p>It is possible that in your case the problem was fixed not because you changed the artifact-id to lowercase, but simply because you changed the artifact-id!</p>
<p>It appears that GitHub has a bug which makes it impossible to publish an artifact if another repository is already publishing an artifact with the same artifact-id. (And then maven screws up on top of that and gives a completely unhelpful error message.) see <a href="https://github.community/t/maven-deploy-fails-with-status-422-unprocessable-entity/253074" class="inline-onebox">Maven deploy fails with status 422 unprocessable entity</a></p>

**Reply by pubudu91** [reply]
Thank you for this! Had the same issue because I had accidentally push the package to the wrong repo. And afterwards, kept getting this 422 error when I tried to push new versions to the correct repo. Deleting the wrongly pushed package worked.

**Comment by anhtester** [comment]
I faced the same problem and haven't found a solution yet.
Someone has solved it please help me.

![image](https://user-images.githubusercontent.com/87883620/211007306-89ff8544-2bdc-4301-aee3-1374b5016b09.png)

**Reply by anhtester** [reply]
Yeah i fixed it when changed artifactId to lowercase.

**Comment by TheLimeGlass** [comment]
I want to add that the jar name of the upload, also must match and be lowercase.
